writer.go

```diff
@@ -0,0 +1,166 @@
+package sequencefile
+
+import (
+	"bytes"
+	"compress/gzip"
+	"crypto/rand"
+	"encoding/binary"
+	"io"
+
+	"github.com/golang/snappy"
+)
+
+type Writer struct {
+	Header        Header
+	writer        io.Writer
+	sinceLastSync int
+	compressor    compressor
+}
+
+func NewWriter(w io.Writer) *Writer {
```

Comment:
This needs a godoc, and the godoc should encourage you to wrap `w` in a `bufio.Writer`, so that the individual writes don't suck.